omc3/plotting/spectrum/waterfall.py

```diff
@@ -87,11 +87,15 @@ def _format_axes(fig_cont, limits, ncol):
     for idx_plane, plane in enumerate(PLANES):
         ax = fig_cont.axes[idx_plane]
         if ncol < 1:
-            ax.set_yticklabels([])
-            ax.set_yticks([])
+            ax.set_yticks(ticks=[], labels=[])
         else:
-            ax.set_yticklabels(ylabels, fontdict={'fontsize': get_fontsize_as_float(matplotlib.rcParams[u'axes.labelsize']) * .5})
-            ax.set_yticks(range(len(ylabels)))
+            # Provide ticks and labels together or matplotlib issues a UserWarning
+            # See "Discouraged" admonition at https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.set_yticklabels.html 
+            ax.set_yticks(
+                ticks=range(len(ylabels)),
+                labels=ylabels,
+                fontdict={'fontsize': get_fontsize_as_float(matplotlib.rcParams[u'axes.labelsize']) * .5},
+            )
```

Comment:
This fixes the `UserWarning: set_ticklabels() should only be used with a fixed number of ticks, i.e. after set_ticks() or using a FixedLocator.`

The recommended way according to the `matplotlib` documentation is to use `axis.set_ticks` and provide both ticks and labels as the same time.